src/validation/manifest.js

```diff
@@ -241,23 +241,61 @@ const validateValue = (value, ctx) => {
   }
 }
 
-const validateDataSourceForNetwork = (dataSources, protocol) =>
+// Transforms list of data sources like this:
+// [
+//   { name: 'contract0', kind: 'ethereum', network: 'mainnet' },
+//   { name: 'contract1', kind: 'ethereum', network: 'mainnet' },
+//   { name: 'contract2', kind: 'ethereum', network: 'xdai' },
+//   { name: 'contract3', kind: 'near', network: 'near-mainnet' },
+// ]
+//
+// Into Immutable JS structure like this:
+// {
+//   ethereum: {
+//     mainnet: ['contract0', 'contract1'],
+//     xdai: ['contract2'],
+//   },
+//   near: {
+//     'near-mainnet': ['contract3'],
+//   },
+// }
+const dataSourceListToMap = dataSources =>
   dataSources
-    .filter(dataSource => protocol.isValidKindName(dataSource.kind))
     .reduce(
-      (networks, dataSource) =>
-        networks.update(dataSource.network, dataSources =>
-          (dataSources || immutable.OrderedSet()).add(dataSource.name),
+      (protocolKinds, dataSource) =>
+        protocolKinds.update(dataSource.kind, networks =>
```

Comment:
Currently if you have two datasources, one with kind ethereum and one with ethereum/contract the validation will fail. Unless this is a desired behaviour, a solution could be to replace dataSource.kind with Protocol.fromDataSources([dataSource]).name

Reply:
Woah, good point! I guess I'll need the `protocol` parameter now 😅 

I'll probably use [Protocol.availableProtocols()](https://github.com/graphprotocol/graph-cli/blob/main/src/protocols/index.js#L25)